### PR TITLE
[docker] fix config bools, skip image stats option

### LINKED ISSUE #1342: High CPU Load  and intermediate image count failure
When there are lot of intermediate docker images, the following line leads to high cpu utilization by docker daemon:

https://github.com/DataDog/dd-agent/blob/5.1.x/checks.d/docker.py#L150

Also the call fails due to timeout .

Can we make intermediate image counter configurable (and turn it off by default ?).  

```
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="GET /containers/json?all=False&size=False"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="+job containers()"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="-job containers() = OK (0)"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="GET /containers/json?all=True&size=False"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="+job containers()"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="-job containers() = OK (0)"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="GET /events?since=1422945112&until=1422945143"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="+job events()"
Feb 03 06:32:23 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:23Z" level="info" msg="-job events() = OK (0)"
Feb 03 06:32:26 ip-10-102-169-13.ec2.internal dockerd[21250]: write unix @: broken pipe
Feb 03 06:32:26 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:26Z" level="info" msg="-job images() = ERR (1)"
Feb 03 06:32:26 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:26Z" level="error" msg="Handler for GET /images/json returned error: write unix @: broken pipe"
Feb 03 06:32:26 ip-10-102-169-13.ec2.internal dockerd[21250]: time="2015-02-03T06:32:26Z" level="error" msg="HTTP Error: statusCode=500 write unix @: broken pipe"
```

**Env Info**:
Docker version: ```Docker version 1.4.1, build 5bc2ff8-dirty```
Agent Image: ``datadog/docker-dd-agent:latest  image-id: 8f65aeca1481``

### PR DESCRIPTION
**Pull request by LeoCavaille**
Fixes #1342.
We should always cast boolean options from the config with
the `_is_affirmative` tool, otherwise you're exposed to
oddities like bool('false') == True.
This introduces a new option `collect_images_stats` that is
enabled by default and skips the collection of metrics
`docker.images.available` and `docker.images.intermediate` that
sometimes is very slow through docker API if you have a lot of
intermediate layer images.

### CODE REVIEW

**Comment by LeoCavaille** [comment]
@remh classifying it as a bugfix still, because of the potential bad boolean flag parsing, so this could still make it for 5.2.

File: conf.d/docker.yaml.example [hunk 1]
```diff
@@ -43,11 +43,12 @@ instances:
     #   - ".*"
     #
     # include all, except ubuntu and debian.
-    # include: []
+    # include: ".*"
```
Comment (LotharSee) [inline conf.d/docker.yaml.example]: Shouldn't it be `[".*"]` instead?

Comment (LeoCavaille) [inline conf.d/docker.yaml.example]: OOPS

Comment (LotharSee) [inline conf.d/docker.yaml.example]: Even worse!
This is wrong, in this case you will include everything.

> How it works: if a tag matches an exclude rule, it won't be included unless it also matches an include rule.

**Comment by LeoCavaille** [comment]
@LotharSee thanks for the review, it was a careless mistake confusing different things, should be gtg now

File: tests/common.py [hunk 2]
```diff
@@ -39,6 +59,7 @@ def load_check(name, config, agentConfig):
     try:
         return check_class(name, init_config=init_config, agentConfig=agentConfig, instances=instances)
     except Exception as e:
+        raise
         raise Exception("Check is using old API, {0}".format(e))
```
Comment (LotharSee) [inline tests/common.py]: One `raise` is enough!

Comment (LeoCavaille) [inline tests/common.py]: :tired_face:

**Comment by LeoCavaille** [comment]
should be fixed now...

**Pull request merged**